fms/training/plugins.py

```diff
@@ -174,22 +169,27 @@ def step(
             steps = step - self.last_reported_step
             self.last_reported_step = step
 
-        world = 1 if self.group is None else self.group.size()
-        if world > 1 and self.tokens_seen.device.type == "cuda":
+        # group and global world are only different for HSDP
+        group_world = 1 if self.group is None else self.group.size()
+        global_world = 1 if not dist.is_initialized() else dist.get_world_size()
+        if global_world > 1 and self.tokens_seen.device.type == "cuda":
```

Comment:
so this makes an assumption that if we do distributed training on cuda devices, then metrics will also be on cuda, right?

I don't see reduce on cpu device, so I assume this has a hidden assumption that if a user train distributed, they cannot put metrics tracker on cpu.

Reply:
FWIW I believe we're doing this in fms-fsdp currently. As far as I can tell the only reason to support cpu training is for unit testing

Reply:
@daviswer this is for token count and accumulated token count, we did it on cpu, right?  we simply do `* world_size` rather than putting it on device and do a reduce.  But fms-fsdp does not support as many scenarios as we support here in fms, thus it was easier in fms-fsdp (e.g. we always assumed cuda and always assumed FSDP.)

Reply:
that's true for the accumulatied metrics (which isn't all metrics). to accumulate cpu metrics we'd have to make sure we're correctly handling with GLOO backend, and I just didn't want to bother testing it since I'm not sure if it's realistic to do meaningful distributed training on CPU